lib/fileprocessor.js

```diff
@@ -151,10 +151,16 @@ var FileProcessor = module.exports = function (type, patterns, finder, logcb, bl
 //
 FileProcessor.prototype.replaceBlocks = function replaceBlocks(file) {
   var result = file.content;
-  var linefeed = /\r\n/g.test(result) ? '\r\n' : '\n';
+
   file.blocks.forEach(function (block) {
-    var blockLine = block.raw.join(linefeed);
-    result = result.replace(blockLine, this.replaceWith(block));
+    var firstBlock = (block.raw[0] + '').replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+    var lastBlock = (block.raw[block.raw.length - 1] + '').replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
```

Comment:
instead of the regexes, just use https://github.com/sindresorhus/escape-string-regexp